lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb

```diff
@@ -45,20 +45,20 @@ def teardown
       global_id_conv = drb_server.config[:idconv]
       drb_server.stop_service
       self.drb_server = nil
+
       # This hack was done to prevent ruby from leaking the
       # TimerIdConv thread.
       # https://bugs.ruby-lang.org/issues/12342 (has been fixed in ruby 2.4.0 preview 1)
       # also fixed in ruby_2_3 branch for the 2.3.2 release: https://github.com/ruby/ruby/commit/c20b07d5357d7cb73226b149431a658cde54a697
-      thread = global_id_conv
-               .try(:instance_variable_get, '@holder')
-               .try(:instance_variable_get, '@keeper')
-      if RUBY_VERSION > "2.3.1"
-        warn "Remove me: #{__FILE__}:#{__LINE__} and my test. Ruby 2.3.2+ should not be creating timer threads."
```

Comment:
Interesting @NickLaMuro @kbrock, especially the 2.5 test.  The problem is we usually support two ruby versions on master, so yeah, we can't remove this hack completely until ruby 2.3.0 and below are unsupported.  

We can as of bundler 1.12+, add
`ruby "~> 2.3", "< 2.4.1"` to the Gemfile, but we can't programmatically access this to check if the current ruby is satisfied.

It would be nice to add a warn if `Bundler.load.required_ruby.satisfied_by?(RUBY_VERSION)` was a thing but the ruby requirement isn't exposed in the bundler runtime...  See https://github.com/bundler/bundler/commit/b4a52c3b29423d3f41c956f638c92ab9830e2716.  Maybe I'm missing something but I couldn't access it.